server/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -400,11 +400,8 @@ public void testDotsWithExistingNestedMapper() throws Exception {
             b.endObject();
         }));
 
-        MapperParsingException e = expectThrows(MapperParsingException.class, () -> mapper.parse(source(b -> b.field("field.bar", 123))));
-        assertEquals(
-            "Cannot add a value for field [field.bar] since one of the intermediate objects is mapped as a nested object: [field]",
-            e.getMessage()
-        );
```

Comment:
A nice upside of this change is that we can now handle dynamically mapped fields with dots that lead into a nested object.